capabilityreceiver/pkg/capabilityhandler/capabilityreceiver.go

```diff
@@ -168,30 +173,57 @@ func saveMsgInDB(message []byte, args *map[string]interface{}) error {
 	} else if err != nil {
 		return err
 	} else {
```

Comment:
I think from 158 - 226 could be refactored some since it looks like there's some repetition with function calls and variables.